### PR TITLE
Fix errors for deployment due to merge conflicts and combining code

### PR DESCRIPTION
**Pull request by ALai2**
The errors due to resolving merge conflicts were fixed and the code was combined to allow the image to be uploaded, similar recipes were found and displayed onto index.html.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/DishAnalysisServlet.java [hunk 1]
```diff
@@ -40,25 +41,26 @@
...
-
-<<<<<<< HEAD:src/main/java/com/google/sps/servlets/DishAnalysisServlet.java
 /** Servlet that uses VisionAPI to analyze uploaded images */
+@MultipartConfig
 @WebServlet("/dishAnalysis")
 public class DishAnalysisServlet extends HttpServlet {
 
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
```
Comment (cocovsack) [inline src/main/java/com/google/sps/servlets/DishAnalysisServlet.java:56]: I really don't understand why those changes weren't saved from before. But yes this looks good!

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -233,25 +231,29 @@ function createRecipeElement(recipe) {
 function createRecipeCardAlerts(recipe) {
   const alertElements = document.createElement('div');
   alertElements.className = 'recipe-card-block';
-  
-  fetch('/profile').then(response => response.json()).then((message) => {
```
Comment (cocovsack) [inline src/main/webapp/script.js:237]: I think we can figure out how to have nested fetches like this but obv we need to find a new approach.

File: src/main/java/com/google/sps/servlets/DishAnalysisServlet.java [hunk 3]
```diff
@@ -73,8 +75,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       List<AnnotateImageResponse> responses = new_response.getResponsesList();
 
       // Initialize blocked catagories
-      Set<String> blockedCatagories = new HashSet<String>;
-      blockedCatagories = new HashSet<String>(Arrays.asList("Cuisine", "Dish", "Food", "Ingredient", "Salad", "Fried food"));
+      Set<String> blockedCatagories = new HashSet<String>(Arrays.asList("Cuisine", "Dish", "Food", "Ingredient", "Salad", "Fried food"));
```
Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/DishAnalysisServlet.java:78]: I think blockedCategories should be constant (private static variable)

Comment (liguanghua) [inline src/main/java/com/google/sps/servlets/DishAnalysisServlet.java:78]: Why not use com.google.common.collect.ImmutableSet?

File: src/main/webapp/script.js [hunk 4]
```diff
@@ -14,11 +14,17 @@
...
-  const request = new Request('/dishAnalysis', {method: "POST"});
+  const image = document.getElementById('image').files[0];
+  const params = new FormData();
+  params.append('image', image);
+  const request = new Request('/dishAnalysis', {method: "POST", body: params});
   fetch(request).then(response => response.json()).then((recipeListInfoJson) => {
+    var recipeList = JSON.parse(JSON.parse(recipeListInfoJson));
     const displayRecipeElement = document.getElementById('display-recipes');
     displayRecipeElement.innerHTML = "";
-    recipeListInfoJson.forEach(recipe => displayRecipeElement.appendChild(createRecipeElement(recipe)));
```
Comment (liguanghua) [inline src/main/webapp/script.js:21]: Just be curious, what's the issue with forEach?

File: src/main/webapp/script.js [hunk 5]
```diff
@@ -242,25 +240,29 @@ function createRecipeElement(recipe) {
...
+  //       // profile.dairyFree
+  //       // profile.allergies
+
+  //       alertElements.appendChild(createAlertElement("icon-warning-sign", "Dietary Alert"));
+  //       alertElements.appendChild(createAlertElement("icon-exclamation", "Dietary Alert"));
+  //       alertElements.appendChild(createAlertElement("icon-leaf", "Non-Vegetarian Alert"));
+  //       alertElements.appendChild(createAlertElement("icon-coffee", "Non-DairyFree Alert"));
+  //       alertElements.appendChild(createAlertElement("icon-food", "Allergies Alert"));
+  //     }
+  //   });
```
Comment (liguanghua) [inline src/main/webapp/script.js:265]: Delete these lines.

File: src/main/webapp/script.js (hunk 4 above)
Comment (ALai2) [reply src/main/webapp/script.js:21]: recipe wasn't giving us back the json that we needed to input into createRecipeElement() when we were using forEach.

File: src/main/java/com/google/sps/servlets/DishAnalysisServlet.java (hunk 3 above)
Comment (ALai2) [reply src/main/java/com/google/sps/servlets/DishAnalysisServlet.java:78]: Using ImmutableSet is a good idea, I added a commit which changes blockedCategories to use ImmutableSet.

**Pull request merged**